File: src/app/model/search-response.ts
import {Attribute} from "./Attribute";
import {Product} from "./Product";
/**
 * User: phi
 * Date: 07.03.17
 * .  .___.
 * .  {o,o}
 * . /)___)
 * . --"-"--
 */
export class SearchResponse {
	products: Product[];
	attributes: Attribute[];

	static fromJson(json: any): SearchResponse  {
		let resp = new SearchResponse();
		let products = Product.fromJsons(json.products);
		let attributes = Attribute.fromJsons(json.attributes);
		resp.products = products;
		resp.attributes = attributes;
		return resp;
	}
}

File: src/main/java/phil294/ls/api/model/SearchResponse.java
package phil294.ls.api.model;

import java.util.List;

/**
 * Tupel Produkte und AttributeOrder
 * User: phi
 * Date: 14.06.17
 * .  .___.
 * .  {o,o}
 * . /)___)
 * . --"-"--
 */
public class SearchResponse
{
	private List<Product> products;
	private List<Integer> attributeOrder;
	
	public List<Product> getProducts()
	{
		return products;
	}
	
	public void setProducts(List<Product> products)
	{
		this.products = products;
	}
	
	public List<Integer> getAttributeOrder()
	{
		return attributeOrder;
	}
	
	public void setAttributeOrder(List<Integer> attributeOrder)
	{
		this.attributeOrder = attributeOrder;
	}
	
	public SearchResponse(List<Product> products, List<Integer> attributeOrder)
	{
		this.products = products;
		this.attributeOrder = attributeOrder;
	}
}
